src/juju/index.js

```diff
@@ -80,12 +87,42 @@ async function loginWithBakery(visitPage, macaroonStore) {
   }
   const juju = await jujulib.connect(
     controllerURL,
-    generateConnectionOptions()
+    generateConnectionOptions(true)
   );
   const conn = await juju.login({});
+  // Ping to keep the connection alive.
+  conn.facades.pinger.pingForever(20000, err => {
```

Comment:
Is it worth checking if pinger is a member of facades here for cases when usePinger is false?

Reply:
For this connection `pinger` should never be undefined as it's included in the generated options above. 

I think I'd prefer it to fail dramatically then silently fail when the pinger doesn't keep the connection alive.